### ISSUE TITLE
Simple example does not work on Ubuntu 14.04 with Docker 1.10.3

### ISSUE THREAD

**Issue by arkadiuszbicz** [jdeathe/centos-ssh#140]
Logs:

root@abksystem:~/dockerbuild# docker run -d \

>   --name ssh.pool-1.1.1 \
>   -p 2020:22 \
>   jdeathe/centos-ssh:centos-7
> 60addcc4d14b505ce5a3fd3e2c86e1058cdefe60e069f37263081828ce46ef39
> root@abksystem:~/dockerbuild# docker logs ssh.pool-1.1.1
> 2016-03-18 10:09:52,713 CRIT Supervisor running as root (no user in config file)
> 2016-03-18 10:09:52,713 WARN Included extra file "/etc/supervisord.d/sshd-bootstrap.conf" during parsing
> 2016-03-18 10:09:52,713 WARN Included extra file "/etc/supervisord.d/sshd.conf" during parsing
> 2016-03-18 10:09:52,713 WARN No file matches via include "/etc/supervisord.d/*.ini"
> 2016-03-18 10:09:52,715 INFO supervisord started with pid 1
> 2016-03-18 10:09:53,719 INFO spawned: 'supervisor_stdout' with pid 14
> 2016-03-18 10:09:53,722 INFO spawned: 'sshd-bootstrap' with pid 15
> 2016-03-18 10:09:53,726 INFO spawned: 'sshd' with pid 16
> 2016-03-18 10:09:53,777 INFO success: sshd-bootstrap entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
> 2016-03-18 10:09:54,771 INFO success: supervisor_stdout entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
> 2016-03-18 10:09:54,772 INFO success: sshd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
> sshd-bootstrap stdout | Initialising SSH.
> sshd-bootstrap stdout | /dev/stdin: Stale file handle
> sshd-bootstrap stdout | /dev/stdin: Stale file handle
> sshd-bootstrap stdout | /dev/stdin: Stale file handle
> sshd-bootstrap stdout | /dev/stdin: Stale file handle
> # sshd-bootstrap stdout | 
> ## SSH Details
> 
> user : app-admin
> password : uWI9JRZ0n6osRgsx
> id : 500:500
> home : /home/app-admin
> chroot path : N/A
> shell : /bin/bash
> sudo : ALL=(ALL) ALL
> key fingerprints :

rsa host key fingerprint :

---

2016-03-18 10:09:55,756 INFO exited: sshd-bootstrap (exit status 0; expected)
root@abksystem:~/dockerbuild# curl -LSs \

>   https://raw.githubusercontent.com/mitchellh/vagrant/master/keys/vagrant \
> 
> > id_rsa_insecure
> > root@abksystem:~/dockerbuild# chmod 600 id_rsa_insecure
> > root@abksystem:~/dockerbuild# ssh -p 2020 -i id_rsa_insecure \
> >   app-admin@172.17.0.1
> > The authenticity of host '[172.17.0.1]:2020 ([172.17.0.1]:2020)' can't be established.
> > ECDSA key fingerprint is b2:52:e7:37:3c:18:00:0f:c5:75:c6:6e:ac:73:63:9a.
> > Are you sure you want to continue connecting (yes/no)? yes
> > Warning: Permanently added '[172.17.0.1]:2020' (ECDSA) to the list of known hosts.
> > Connection to 172.17.0.1 closed.
> > root@abksystem:~/dockerbuild# docker logs ssh.pool-1.1.12016-03-18 10:09:52,713 CRIT Supervisor running as root (no user in config file)
> > 2016-03-18 10:09:52,713 WARN Included extra file "/etc/supervisord.d/sshd-bootstrap.conf" during parsing
> > 2016-03-18 10:09:52,713 WARN Included extra file "/etc/supervisord.d/sshd.conf" during parsing
> > 2016-03-18 10:09:52,713 WARN No file matches via include "/etc/supervisord.d/*.ini"
> > 2016-03-18 10:09:52,715 INFO supervisord started with pid 1
> > 2016-03-18 10:09:53,719 INFO spawned: 'supervisor_stdout' with pid 14
> > 2016-03-18 10:09:53,722 INFO spawned: 'sshd-bootstrap' with pid 15
> > 2016-03-18 10:09:53,726 INFO spawned: 'sshd' with pid 16
> > 2016-03-18 10:09:53,777 INFO success: sshd-bootstrap entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
> > 2016-03-18 10:09:54,771 INFO success: supervisor_stdout entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
> > 2016-03-18 10:09:54,772 INFO success: sshd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
> > sshd-bootstrap stdout | Initialising SSH.
> > sshd-bootstrap stdout | /dev/stdin: Stale file handle
> > sshd-bootstrap stdout | /dev/stdin: Stale file handle
> > sshd-bootstrap stdout | /dev/stdin: Stale file handle
> > sshd-bootstrap stdout | /dev/stdin: Stale file handle
> > # sshd-bootstrap stdout | 
> > ## SSH Details
> > 
> > user : app-admin
> > password : uWI9JRZ0n6osRgsx
> > id : 500:500
> > home : /home/app-admin
> > chroot path : N/A
> > shell : /bin/bash
> > sudo : ALL=(ALL) ALL
> > key fingerprints :

rsa host key fingerprint :

---

2016-03-18 10:09:55,756 INFO exited: sshd-bootstrap (exit status 0; expected)
sshd stdout | Server listening on 0.0.0.0 port 22.
Server listening on :: port 22.
sshd stdout | Accepted publickey for app-admin from 172.17.0.1 port 49521 ssh2: RSA dd:3b:b8:2e:85:04:06:e9:ab:ff:a8:0a:c0:04:6e:d6
sshd stdout | PAM: pam_open_session(): Cannot make/remove an entry for the specified session
sshd stdout | lastlog_openseek: Couldn't stat /var/log/lastlog: No such file or directory
sshd stdout | lastlog_openseek: Couldn't stat /var/log/lastlog: No such file or directory
sshd stdout | syslogin_perform_logout: logout() returned an error
sshd stdout | Received disconnect from 172.17.0.1: 11: disconnected by user
2016-03-18 10:10:46,709 INFO reaped unknown pid 110
`

**Comment by jdeathe** [comment]
Hi @arkadiuszbicz 

Please confirm what exact OS and version you are using. Is it Debian or Ubuntu?

**Comment by jdeathe** [comment]
Hi @arkadiuszbicz 

I have reproduced this issue on Ubuntu 14.0.4-LTS + Docker 1.10.3

I develop/test using docker-machine / CoreOS / CentOS-7 as the Docker host.

**Comment by jdeathe** [comment]
Also working on Fedora 23 + Docker 1.9.1

**Comment by jdeathe** [comment]
@arkadiuszbicz 

Initial investigation suggests this issue occurs on Docker 1.10 so using a docker host with 1.9 should work until I have a fix for this issue.

I have also reproduced it with Kitematic which is using 1.10.2 on my environment.

**Comment by jdeathe** [comment]
@arkadiuszbicz 

Another work-around appears to be to add your SSH public key via the `SSH_AUTHORIZED_KEYS` environment variable. If using the Vagrant insecure private key then you would need to add the public key from: https://raw.githubusercontent.com/mitchellh/vagrant/master/keys/vagrant.pub

The run command would need replacing with:

```
$ docker run -d \
--name ssh.pool-1.1.1 \
-p 2020:22 \
-e "SSH_AUTHORIZED_KEYS=ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ==" \
jdeathe/centos-ssh:centos-7
```

**Comment by arkadiuszbicz** [comment]
Still the same error, I have used my own public key.

sshd stdout | Server listening on 0.0.0.0 port 22.
Server listening on :: port 22.
sshd stdout | Connection closed by 172.17.0.1 [preauth]
2016-03-18 15:30:22,354 INFO reaped unknown pid 107
sshd stdout | Connection closed by 172.17.0.1 [preauth]
2016-03-18 15:31:21,141 INFO reaped unknown pid 109
sshd stdout | Connection closed by 172.17.0.1 [preauth]
2016-03-18 15:31:34,173 INFO reaped unknown pid 111
sshd stdout | Connection closed by 172.17.0.1 [preauth]
2016-03-18 15:31:43,180 INFO reaped unknown pid 113
